### ISSUE TITLE
N-Body Benchmark

### ISSUE THREAD

**Issue by nidin** [dcodeIO/AssemblyScript#14]
We already have an n-body benchmark is it ready to compile with AssemblyScript?

**Comment by MaxGraey** [comment]
Still needs generics

**Comment by dcodeIO** [comment]
The benchmark, for reference: https://github.com/MaxGraey/TurboScript/blob/MaxGraey/add-benchmark/benchmark/n-body/ts/index.ts

**Comment by MaxGraey** [comment]
Yes, you can refine code for better compatibility with AssemblyScript if need

**Comment by dcodeIO** [comment]
For reference:
* [nbody.ts](https://github.com/dcodeIO/AssemblyScript/blob/master/tests/nbody.ts)
* [nbody.wast](https://github.com/dcodeIO/AssemblyScript/blob/master/tests/nbody.wast)
* [nbody.wasm](https://github.com/dcodeIO/AssemblyScript/blob/master/tests/nbody.wasm)

Still needs working arrays, though.

**Comment by dcodeIO** [comment]
I've managed to compile the benchmark test case without issues now, but it's still trapping with an out of range error when executing `test`. If anyone is interested to investigate this further, that'd help to get the benchmark up and running.

**Comment by dcodeIO** [comment]
With the latest commit, it now measures *something* and doesn't explode anymore, but the return value still seems to be bogus. To try it yourself, open `tests/index.html` (FF works) and load `tests/benchmarks/nbody.wasm` with it. Executing `exports.test(N_ITERATIONS)` in the browser console then runs it. From what I can tell, the return value (energy) becomes lower and lower until it wraps to NaN. Unless that's intended, there's still something strange going on (any help appreciated).
